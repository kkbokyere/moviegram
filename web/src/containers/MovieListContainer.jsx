import React, { Component } from 'react';
import PropTypes from 'prop-types'
import MovieList from "../components/MovieList";
import {connect} from 'react-redux';
import {getAllMovies} from "../actions/moviesActions";

class MovieListContainer extends Component {
    constructor(props) {
        super(props);
    }
    componentWillMount() {
        if(!this.props.movies.length) {
            this.props.getAllMovies();
        }
    }
    render() {
        return(<MovieList data={this.props.movies}/>)
    }
}

const mapStateToProps = (state) => {
    return {
        movies: state.movies.data
    }
};

const mapDispatchToProps = (dispatch) => {
    return {
        getAllMovies: () => dispatch(getAllMovies()),
    }
};


MovieListContainer.propTypes = {
    movies: PropTypes.array,
    getAllMovies: PropTypes.func,
};

export default connect(mapStateToProps, mapDispatchToProps)(MovieListContainer);