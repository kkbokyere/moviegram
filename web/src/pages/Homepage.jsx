import React from 'react';
import MovieListContainer from "../containers/MovieListContainer";

const Homepage = ()=> {
    return(<MovieListContainer/>);
};

export default Homepage;