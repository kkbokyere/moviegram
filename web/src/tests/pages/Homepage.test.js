import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';

import Homepage from '../../pages/Homepage';
import MovieListContainer from '../../containers/MovieListContainer';

describe('Homepage Component', () => {
    it('should render Homepage', () => {
        const wrapper = renderer.create(<Homepage />).toJSON();
        expect(wrapper).toMatchSnapshot()
    });
    it('should render Homepage with MovieListContainer', () => {
        const wrapper = mount(<Homepage />);
        expect(wrapper.find(MovieListContainer).length).toEqual(1)
    });
});