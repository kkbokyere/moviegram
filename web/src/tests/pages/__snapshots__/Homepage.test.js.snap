// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Component should render Homepage 1`] = `
<section
  className="movie-list"
>
  <article
    className="movie-item"
  >
    <div
      className="movie-item__img"
    >
      <picture>
        <source
          media="(min-width: 768px)"
          srcSet="https://via.placeholder.com/200x300"
        />
        <img
          alt="AVENGERS: ENDGAME"
          src="https://via.placeholder.com/400x300"
        />
      </picture>
    </div>
    <div
      className="movie-item__wrapper"
    >
      <h1
        className="movie-item__title"
      >
        AVENGERS: ENDGAME
      </h1>
      <div
        className="movie-item__showtimes"
      >
        <h3>
          Show Times:
        </h3>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Wood Green - Vue
        </span>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Enfield - Cineworld
        </span>
      </div>
      <button
        className="btn btn-primary"
      >
        See more
      </button>
    </div>
  </article>
  <article
    className="movie-item"
  >
    <div
      className="movie-item__img"
    >
      <picture>
        <source
          media="(min-width: 768px)"
          srcSet="https://via.placeholder.com/200x300"
        />
        <img
          alt="POKEMON: DETECTIVE PIKACHU"
          src="https://via.placeholder.com/400x300"
        />
      </picture>
    </div>
    <div
      className="movie-item__wrapper"
    >
      <h1
        className="movie-item__title"
      >
        POKEMON: DETECTIVE PIKACHU
      </h1>
      <div
        className="movie-item__showtimes"
      >
        <h3>
          Show Times:
        </h3>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Wood Green - Vue
        </span>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Enfield - Cineworld
        </span>
      </div>
      <button
        className="btn btn-primary"
      >
        See more
      </button>
    </div>
  </article>
  <article
    className="movie-item"
  >
    <div
      className="movie-item__img"
    >
      <picture>
        <source
          media="(min-width: 768px)"
          srcSet="https://via.placeholder.com/200x300"
        />
        <img
          alt="JOHN WICK: CHAPTER 3 - PARABELLUM"
          src="https://via.placeholder.com/400x300"
        />
      </picture>
    </div>
    <div
      className="movie-item__wrapper"
    >
      <h1
        className="movie-item__title"
      >
        JOHN WICK: CHAPTER 3 - PARABELLUM
      </h1>
      <div
        className="movie-item__showtimes"
      >
        <h3>
          Show Times:
        </h3>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Wood Green - Vue
        </span>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Enfield - Cineworld
        </span>
      </div>
      <button
        className="btn btn-primary"
      >
        See more
      </button>
    </div>
  </article>
  <article
    className="movie-item"
  >
    <div
      className="movie-item__img"
    >
      <picture>
        <source
          media="(min-width: 768px)"
          srcSet="https://via.placeholder.com/200x300"
        />
        <img
          alt="THE ROYAL BALLET: TRIPLE BILL (2019)"
          src="https://via.placeholder.com/400x300"
        />
      </picture>
    </div>
    <div
      className="movie-item__wrapper"
    >
      <h1
        className="movie-item__title"
      >
        THE ROYAL BALLET: TRIPLE BILL (2019)
      </h1>
      <div
        className="movie-item__showtimes"
      >
        <h3>
          Show Times:
        </h3>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Wood Green - Vue
        </span>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Enfield - Cineworld
        </span>
      </div>
      <button
        className="btn btn-primary"
      >
        See more
      </button>
    </div>
  </article>
  <article
    className="movie-item"
  >
    <div
      className="movie-item__img"
    >
      <picture>
        <source
          media="(min-width: 768px)"
          srcSet="https://via.placeholder.com/200x300"
        />
        <img
          alt="PAW PATROL MIGHTY PUPS: THE MOVIE"
          src="https://via.placeholder.com/400x300"
        />
      </picture>
    </div>
    <div
      className="movie-item__wrapper"
    >
      <h1
        className="movie-item__title"
      >
        PAW PATROL MIGHTY PUPS: THE MOVIE
      </h1>
      <div
        className="movie-item__showtimes"
      >
        <h3>
          Show Times:
        </h3>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Barnet - Everyman
        </span>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Finchley - Cineworld
        </span>
      </div>
      <button
        className="btn btn-primary"
      >
        See more
      </button>
    </div>
  </article>
</section>
`;

exports[`Homepage Component should render MovieList 1`] = `
<section
  className="movie-list"
>
  <article
    className="movie-item"
  >
    <div
      className="movie-item__img"
    >
      <picture>
        <source
          media="(min-width: 768px)"
          srcSet="https://via.placeholder.com/200x300"
        />
        <img
          alt="AVENGERS: ENDGAME"
          src="https://via.placeholder.com/400x300"
        />
      </picture>
    </div>
    <div
      className="movie-item__wrapper"
    >
      <h1
        className="movie-item__title"
      >
        AVENGERS: ENDGAME
      </h1>
      <div
        className="movie-item__showtimes"
      >
        <h3>
          Show Times:
        </h3>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Wood Green - Vue
        </span>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Enfield - Cineworld
        </span>
      </div>
      <button
        className="btn btn-primary"
      >
        See more
      </button>
    </div>
  </article>
  <article
    className="movie-item"
  >
    <div
      className="movie-item__img"
    >
      <picture>
        <source
          media="(min-width: 768px)"
          srcSet="https://via.placeholder.com/200x300"
        />
        <img
          alt="POKEMON: DETECTIVE PIKACHU"
          src="https://via.placeholder.com/400x300"
        />
      </picture>
    </div>
    <div
      className="movie-item__wrapper"
    >
      <h1
        className="movie-item__title"
      >
        POKEMON: DETECTIVE PIKACHU
      </h1>
      <div
        className="movie-item__showtimes"
      >
        <h3>
          Show Times:
        </h3>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Wood Green - Vue
        </span>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Enfield - Cineworld
        </span>
      </div>
      <button
        className="btn btn-primary"
      >
        See more
      </button>
    </div>
  </article>
  <article
    className="movie-item"
  >
    <div
      className="movie-item__img"
    >
      <picture>
        <source
          media="(min-width: 768px)"
          srcSet="https://via.placeholder.com/200x300"
        />
        <img
          alt="JOHN WICK: CHAPTER 3 - PARABELLUM"
          src="https://via.placeholder.com/400x300"
        />
      </picture>
    </div>
    <div
      className="movie-item__wrapper"
    >
      <h1
        className="movie-item__title"
      >
        JOHN WICK: CHAPTER 3 - PARABELLUM
      </h1>
      <div
        className="movie-item__showtimes"
      >
        <h3>
          Show Times:
        </h3>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Wood Green - Vue
        </span>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Enfield - Cineworld
        </span>
      </div>
      <button
        className="btn btn-primary"
      >
        See more
      </button>
    </div>
  </article>
  <article
    className="movie-item"
  >
    <div
      className="movie-item__img"
    >
      <picture>
        <source
          media="(min-width: 768px)"
          srcSet="https://via.placeholder.com/200x300"
        />
        <img
          alt="THE ROYAL BALLET: TRIPLE BILL (2019)"
          src="https://via.placeholder.com/400x300"
        />
      </picture>
    </div>
    <div
      className="movie-item__wrapper"
    >
      <h1
        className="movie-item__title"
      >
        THE ROYAL BALLET: TRIPLE BILL (2019)
      </h1>
      <div
        className="movie-item__showtimes"
      >
        <h3>
          Show Times:
        </h3>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Wood Green - Vue
        </span>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Enfield - Cineworld
        </span>
      </div>
      <button
        className="btn btn-primary"
      >
        See more
      </button>
    </div>
  </article>
  <article
    className="movie-item"
  >
    <div
      className="movie-item__img"
    >
      <picture>
        <source
          media="(min-width: 768px)"
          srcSet="https://via.placeholder.com/200x300"
        />
        <img
          alt="PAW PATROL MIGHTY PUPS: THE MOVIE"
          src="https://via.placeholder.com/400x300"
        />
      </picture>
    </div>
    <div
      className="movie-item__wrapper"
    >
      <h1
        className="movie-item__title"
      >
        PAW PATROL MIGHTY PUPS: THE MOVIE
      </h1>
      <div
        className="movie-item__showtimes"
      >
        <h3>
          Show Times:
        </h3>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Barnet - Everyman
        </span>
        <span
          className="movie-item__time"
        >
          May 16th 2019, 3:24:25 pm
           @ 
          Finchley - Cineworld
        </span>
      </div>
      <button
        className="btn btn-primary"
      >
        See more
      </button>
    </div>
  </article>
</section>
`;
